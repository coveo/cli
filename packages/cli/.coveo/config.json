{"version": 1, "organization": ""}
