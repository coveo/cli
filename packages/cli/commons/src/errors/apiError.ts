import {red} from 'chalk';
import dedent from 'ts-dedent';
import {PrintableError, SeverityLevel} from './printableError';

export interface APIErrorResponse {
  message: string;
  errorCode: string;
  requestID: string;
}

export interface AxiosErrorFromAPI {
  response: {
    status: number;
    data: APIErrorResponse;
  };
}

export const APIErrorSchema = {
  type: 'object',
  required: ['errorCode', 'message'],
  properties: {
    errorCode: {
      type: 'string',
    },
    message: {
      type: 'string',
    },
  },
};

export const AxiosErrorFromAPISchema = {
  type: 'object',
  required: ['response'],
  properties: {
    response: {
      type: 'object',
      required: ['status', 'data'],
      properties: {
        status: {
          type: 'integer',
        },
        data: APIErrorSchema,
      },
    },
  },
};

export class APIError extends PrintableError {
  public constructor(
    error: APIErrorResponse | AxiosErrorFromAPI,
    tagLine?: string
  ) {
    super(SeverityLevel.Error);
    let errorCode: string;
    let message: string;
    let status: number | undefined;
    if (this.isFromAxios(error)) {
      errorCode = error.response.data.errorCode;
      message = error.response.data.message;
      status = error.response.status;
    } else {
      errorCode = error.errorCode;
      message = error.message;
    }

    this.name = `APIError - ${errorCode}`;
    this.message = dedent`
    ${tagLine}
    ${status ? `Status code: ${status}\n` : ''}
    Error code: ${red(errorCode)}
    Message: ${red(message)}
    `;
  }

  private isFromAxios(
    error: APIErrorResponse | AxiosErrorFromAPI
  ): error is AxiosErrorFromAPI {
    return 'response' in error;
  }
}
