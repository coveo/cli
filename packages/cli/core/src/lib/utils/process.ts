import {spawn, SpawnOptions, ChildProcess} from 'child_process';

function isErrnoException(error: Error): error is NodeJS.ErrnoException {
  return Object.hasOwnProperty.call(error, 'errno');
}

/**
 *
 * @private
 * @param {string} command The command to run
 * @param {string[]} args List of string arguments
 * @param {SpawnOptions} [options={}]
 * @returns {Promise<number>} A promise of the the child process exit code
 */
export function spawnProcess(
  command: string,
  args: string[],
  options: SpawnOptions = {}
): Promise<number> {
  return new Promise((resolve, reject) => {
    spawn(command, args, {
      stdio: ['inherit', 'inherit', 'inherit'],
      ...options,
    }).on('close', (code) => {
      if (code !== 0) {
        // TODO: inject logger decorator so this util method can log
        // this.debug(
        //   `Following command has failed: ${command} ${args.join(' ')}`
        // );

        // Any error message triggered by the execution of the subprocess will be
        // displayed since the terminal is shared by both parent and child processes.
        // So no need to bubble up the error.
        reject(code);
      } else {
        resolve(0);
      }
    });
  });
}

export interface SpawnProcessOutput {
  stdout: string;
  stderr: string;
  exitCode?: string;
}

export async function spawnProcessOutput(
  command: string,
  args: string[],
  options: SpawnOptions = {}
): Promise<SpawnProcessOutput> {
  return new Promise((resolve) => {
    const output: SpawnProcessOutput = {
      stdout: '',
      stderr: '',
    };

    const child = spawn(command, args, options);

    child.stdout?.on('data', (d) => {
      output.stdout += d;
    });

    child.stderr?.on('data', (d) => {
      output.stderr += d;
    });

    const closeListener = (code: number | null) => {
      output.exitCode = Number.isInteger(code) ? String(code) : undefined;
      resolve(output);
    };

    child.on('close', closeListener);

    child.on('error', (e) => {
      output.stderr += e.message;
      if (isErrnoException(e) && e.code) {
        child.off('close', closeListener);
        output.exitCode = e.code;
        resolve(output);
      }
    });
  });
}

export const handleForkedProcess = (subprocess: ChildProcess) => {
  return new Promise<void>((resolve, reject) => {
    subprocess.on('exit', () => {
      resolve();
    });
    subprocess.on('message', (message) => {
      reject(message);
    });
  });
};
