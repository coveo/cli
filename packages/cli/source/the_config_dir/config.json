{"version": "1.1.0", "environment": "prod", "region": "us", "organization": ""}
