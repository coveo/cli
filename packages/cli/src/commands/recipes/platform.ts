import {bold} from 'chalk';
import {Command} from '@oclif/core';

export default class PlatformRecipe extends Command {
  public static description = `${bold.bgYellow(
    '(alpha)'
  )} Run a platform recipe`;

  public static flags = {};

  public static args = [];

  public async run() {
    this.warn('Not implemented yet');
  }

  public async catch(err?: Error & {exitCode?: number}) {
    throw err;
  }
}
