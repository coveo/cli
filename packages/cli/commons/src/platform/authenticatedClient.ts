import {setGlobalDispatcher, ProxyAgent, FormData, fetch} from 'undici';
import PlatformClient from '@coveo/platform-client';
import {Config, Configuration} from '../config/config';
import {castEnvironmentToPlatformClient} from './environment';
import globalConfig from '../config/globalConfig';
import {lt} from 'semver';

export class AuthenticatedClient {
  public cfg: Config;
  public constructor() {
    this.cfg = new Config(globalConfig.get().configDir);
  }

  public async isLoggedIn() {
    const {accessToken} = this.cfg.get();
    return accessToken !== undefined;
  }

  public async isExpired() {
    try {
      const c = await this.getClient();
      await c.initialize();
      return false;
    } catch (e) {
      return true;
    }
  }

  public async getClient(overrideConfig?: Partial<Configuration>) {
    const configFromDisk = this.cfg.get();
    const resolvedConfig = {...configFromDisk, ...overrideConfig};
    const globalRequestSettings: Record<string, unknown> = {};
    const proxyServer =
      process.env['https_proxy'] || process.env['HTTPS_PROXY'];
    if (proxyServer) {
      setGlobalDispatcher(new ProxyAgent(proxyServer));
    }
    if (lt(process.version, '18.0.0')) {
      Object.assign(global, {FormData, fetch});
    }
    return new PlatformClient({
      globalRequestSettings,
      environment: castEnvironmentToPlatformClient(resolvedConfig.environment),
      region: resolvedConfig.region,
      organizationId: resolvedConfig.organization,
      accessToken: resolvedConfig.accessToken!,
    });
  }

  public async getAllOrgsUserHasAccessTo() {
    const platformClient = await this.getClient();
    return platformClient.organization.list();
  }

  public async getUserHasAccessToOrg(org: string) {
    const orgs = await this.getAllOrgsUserHasAccessTo();
    return orgs.some((o) => o.id === org);
  }

  public async createImpersonateApiKey(name: string) {
    const platformClient = await this.getClient();
    return await platformClient.apiKey.create({
      displayName: `cli-${name}`,
      description: 'Generated by the Coveo CLI',
      enabled: true,
      privileges: [
        {targetDomain: 'IMPERSONATE', targetId: '*', owner: 'SEARCH_API'},
      ],
    });
  }

  public async getUsername() {
    const {anonymous, organization} = this.cfg.get();
    if (anonymous) {
      return `api-key@${organization}`;
    }
    const authenticatedClient = new AuthenticatedClient();
    const platformClient = await authenticatedClient.getClient();
    await platformClient.initialize();

    return (await platformClient.user.get()).providerUsername;
  }
}

export enum AuthenticationStatus {
  LOGGED_IN,
  EXPIRED,
  LOGGED_OUT,
}

export async function getAuthenticationStatus() {
  const authenticatedClient = new AuthenticatedClient();

  if (!(await authenticatedClient.isLoggedIn())) {
    return AuthenticationStatus.LOGGED_OUT;
  }

  if (await authenticatedClient.isExpired()) {
    return AuthenticationStatus.EXPIRED;
  }

  return AuthenticationStatus.LOGGED_IN;
}
