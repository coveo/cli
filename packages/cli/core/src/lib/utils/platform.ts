import {Config} from '@coveo/cli-commons/config/config';

export function getTargetOrg(config: Config, target?: string) {
  if (target) {
    return target;
  }
  const cfg = config.get();
  return cfg.organization;
}
