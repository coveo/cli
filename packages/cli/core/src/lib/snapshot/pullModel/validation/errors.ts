import {ValidationError} from 'jsonschema';
import {CLIBaseError} from '@coveo/cli-commons/errors/cliBaseError';
import {getPrettyJsonValidationErrors} from '@coveo/cli-commons/errors/jsonError';

abstract class BaseSPMError extends CLIBaseError {
  public name = 'Snapshot Pull Model Error';
  private static messageSuffix =
    '\nThis is probably a problem with the Coveo CLI, please report this issue at https://github.com/coveo/cli/issues';
  public constructor(message: string, shouldContactCoveo: boolean) {
    super(`${message}${shouldContactCoveo ? BaseSPMError.messageSuffix : ''}`);
  }
}

export class UnknownSPMValidationError extends BaseSPMError {
  public name = 'Unknown Snapshot Pull Model Error';
  public constructor(shouldContactCoveo: boolean) {
    super(
      "An unknown error occured while validating the custom template. Try recreating it from 'empty' or 'full'.",
      shouldContactCoveo
    );
  }
}

export class InvalidSPMError extends BaseSPMError {
  public name = 'Invalid Snapshot Pull Model Error';
  public constructor(shouldContactCoveo: boolean, errors: ValidationError[]) {
    super(getPrettyJsonValidationErrors(errors), shouldContactCoveo);
  }
}
