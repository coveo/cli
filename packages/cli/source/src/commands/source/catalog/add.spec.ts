jest.mock('@coveo/cli-commons/analytics/amplitudeClient');

jest.mock('@coveo/cli-commons/platform/authenticatedClient');
jest.mock('@coveo/cli-commons/config/config');
jest.mock('@coveo/cli-commons/config/globalConfig');
jest.mock('@coveo/cli-commons/preconditions/trackable');
jest.mock('@coveo/push-api-client');

import stripAnsi from 'strip-ansi';
import {test} from '@oclif/test';
import {AuthenticatedClient} from '@coveo/cli-commons/platform/authenticatedClient';
import {CatalogSource, DocumentBuilder} from '@coveo/push-api-client';
import {cwd} from 'process';
import {join} from 'path';
import {APIError} from '@coveo/cli-commons/errors/apiError';
import {Interfaces} from '@oclif/core';
import globalConfig from '@coveo/cli-commons/config/globalConfig';
import {Config} from '@coveo/cli-commons/config/config';

import {
  BatchUploadDocumentsError,
  BatchUploadDocumentsSuccess,
} from '../../../lib/testsUtils/batchUploadDocumentsFromFilesReturn';
const mockedGlobalConfig = jest.mocked(globalConfig);
const mockedClient = jest.mocked(AuthenticatedClient);
const mockedSource = jest.mocked(CatalogSource);
const mockedDocumentBuilder = jest.mocked(DocumentBuilder);
const mockedMarshal = jest.fn();
const mockEvaluate = jest.fn();
const mockSourceGet = jest.fn();
const mockSetSourceStatus = jest.fn();
const mockBatchUpdate = jest.fn();
const mockBatchStream = jest.fn();
const mockedConfigGet = jest.fn();
const mockedConfig = jest.mocked(Config, true);

describe('source:catalog:add', () => {
  const pathToStub = join(cwd(), 'src', '__stub__');

  const sourceContainsDocuments = () => {
    mockSourceGet.mockResolvedValue({
      information: {numberOfDocuments: 1},
    });
  };

  const sourceDoesNotContainDocuments = () => {
    mockSourceGet.mockResolvedValue({
      information: {numberOfDocuments: 0},
    });
  };

  const mockUserHavingAllRequiredPlatformPrivileges = () => {
    mockEvaluate.mockResolvedValue({approved: true});
  };

  const mockUserNotHavingAllRequiredPlatformPrivileges = () => {
    mockEvaluate.mockResolvedValue({approved: false});
  };

  const doMockSuccessUpload = () => {
    mockBatchUpdate.mockReturnValue(new BatchUploadDocumentsSuccess());
    mockBatchStream.mockReturnValue(new BatchUploadDocumentsSuccess());
  };

  const doMockErrorUpload = () => {
    mockBatchUpdate.mockReturnValue(new BatchUploadDocumentsError());
    mockBatchStream.mockReturnValue(new BatchUploadDocumentsError());
  };

  const doMockDocumentBuilder = () => {
    mockedDocumentBuilder.mockImplementation(
      () =>
        ({
          marshal: mockedMarshal,
          withData: jest.fn(),
          withDate: jest.fn(),
          withFileExtension: jest.fn(),
          withMetadataValue: jest.fn(),
        } as unknown as DocumentBuilder)
    );
  };
  const doMockAuthenticatedClient = () => {
    mockedClient.mockImplementation(
      () =>
        ({
          getClient: () =>
            Promise.resolve({
              privilegeEvaluator: {
                evaluate: mockEvaluate,
              },
              source: {
                get: mockSourceGet,
              },
            }),
          cfg: {
            get: () => ({
              accessToken: 'the_token',
              organization: 'the_org',
              region: 'au',
              environment: 'prod',
            }),
          },
        } as unknown as AuthenticatedClient)
    );
  };

  const doMockSource = () => {
    mockedSource.mockImplementation(
      () =>
        ({
          batchUpdateDocumentsFromFiles: mockBatchUpdate,
          batchStreamDocumentsFromFiles: mockBatchStream,
          setSourceStatus: mockSetSourceStatus,
        } as unknown as CatalogSource)
    );
  };

  const doMockConfig = () => {
    mockedGlobalConfig.get.mockReturnValue({
      configDir: 'the_config_dir',
      version: '1.2.3',
      platform: 'darwin',
    } as Interfaces.Config);
    mockedConfigGet.mockReturnValue({
      region: 'us',
      organization: 'default-org',
      environment: 'prod',
    });

    mockedConfig.mockImplementation(
      () =>
        ({
          get: mockedConfigGet,
        } as unknown as Config)
    );
  };

  beforeAll(() => {
    doMockConfig();
    doMockDocumentBuilder();
    doMockAuthenticatedClient();
    doMockSource();
  });

  beforeEach(() => {
    mockedMarshal.mockReturnValue(
      JSON.stringify({
        documentId: 'https://perdu.com',
        title: 'hello world',
      })
    );
  });

  mockedSource.mockImplementation(
    () =>
      ({
        batchUpdateDocumentsFromFiles: mockBatchUpdate,
        setSourceStatus: mockSetSourceStatus,
      } as unknown as CatalogSource)
  );

  describe('when the batch upload is successfull', () => {
    beforeAll(() => {
      mockUserHavingAllRequiredPlatformPrivileges();
      doMockSuccessUpload();
      sourceContainsDocuments();
    });

    afterAll(() => {
      mockBatchUpdate.mockReset();
      mockEvaluate.mockReset();
      mockSourceGet.mockReset();
    });

    test
      .stdout()
      .stderr()
      .command(['source:catalog:add', 'mysource'])
      .catch(/You must set the `files` flag/)
      .it('throws when no flags are specified');

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should trigger a batch document update', () => {
        expect(mockBatchUpdate).toHaveBeenCalled();
        expect(mockBatchStream).not.toHaveBeenCalled();
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '--fullUpload',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should trigger a batch stream upload ', () => {
        expect(mockBatchStream).toHaveBeenCalled();
        expect(mockBatchUpdate).not.toHaveBeenCalled();
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('pass correct configuration information to push-api-client', () => {
        expect(mockedSource).toHaveBeenCalledWith('the_token', 'the_org', {
          environment: 'prod',
          region: 'au',
        });
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should create missing fields by default', () => {
        expect(mockBatchUpdate).toHaveBeenCalledWith(
          expect.anything(),
          expect.anything(),
          expect.objectContaining({createFields: true})
        );
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '--no-createMissingFields',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should skip field creation if specified', () => {
        expect(mockBatchUpdate).toHaveBeenCalledWith(
          expect.anything(),
          expect.anything(),
          expect.objectContaining({createFields: false})
        );
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments', 'someJsonFile.json'),
      ])
      .it('should output feedback message when uploading documents', (ctx) => {
        expect(ctx.stdout).toContain(
          'Success: 2 documents accepted by the API from'
        );
        expect(ctx.stdout).toContain('Status code: 202 ðŸ‘Œ');
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments'),
      ])
      .it(
        'should output feedback message when uploading a directory',
        (ctx) => {
          expect(ctx.stdout).toContain(
            'Success: 2 documents accepted by the API from'
          );
          expect(ctx.stdout).toContain('Status code: 202 ðŸ‘Œ');
        }
      );

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should get source info during document update', () => {
        expect(mockSourceGet).toHaveBeenCalled();
      });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '--fullUpload',
        '-f',
        join(pathToStub, 'jsondocuments', 'batman.json'),
      ])
      .it('should not get source info during document update', () => {
        expect(mockSourceGet).not.toHaveBeenCalled();
      });

    describe('when the source does not contain items', () => {
      beforeAll(() => {
        mockSourceGet.mockReset();
        sourceDoesNotContainDocuments();
      });

      test
        .stdout()
        .stderr()
        .command([
          'source:catalog:add',
          'mysource',
          '-f',
          join(pathToStub, 'jsondocuments', 'batman.json'),
        ])
        .catch(/No items detected for this source at the moment/)
        .it('should show error message during document update');

      test
        .stdout()
        .stderr()
        .command([
          'source:catalog:add',
          'mysource',
          '--skipFullUploadCheck',
          '-f',
          join(pathToStub, 'jsondocuments', 'batman.json'),
        ])
        .it(
          'should not show a error message when using the --skipFullUploadCheck flag',
          (ctx) => {
            expect(ctx.stderr).not.toContain(
              'please consider doing a full catalog upload'
            );
          }
        );
    });
  });

  describe('when the batch upload fails', () => {
    beforeAll(() => {
      doMockErrorUpload();
      mockUserHavingAllRequiredPlatformPrivileges();
      sourceContainsDocuments();
    });

    afterAll(() => {
      mockBatchUpdate.mockReset();
      mockEvaluate.mockReset();
      mockSourceGet.mockReset();
    });

    test
      .stdout()
      .stderr()
      .command([
        'source:catalog:add',
        'mysource',
        '-f',
        join(pathToStub, 'jsondocuments'),
      ])
      .catch((error) => {
        expect(error).toBeInstanceOf(APIError);
        const message = stripAnsi(error.message);
        expect(message).toContain(
          'this is a bad request and you should feel bad'
        );
        expect(message).toContain('Status code: 412');
        expect(message).toContain('Error code: BAD_REQUEST');
      })
      .it('returns an information message on add failure from the API');
  });

  describe('when Platform privilege preconditions are not respected', () => {
    beforeEach(() => {
      mockUserNotHavingAllRequiredPlatformPrivileges();
      sourceContainsDocuments();
    });

    afterAll(() => {
      mockBatchUpdate.mockReset();
      mockEvaluate.mockReset();
      mockSourceGet.mockReset();
    });

    test
      .stdout()
      .stderr()
      .command(['source:catalog:add', 'some-org', '-f', 'some-file'])
      .catch(/You are not authorized to create or update fields/)
      .it('should return a precondition error');
  });
});
