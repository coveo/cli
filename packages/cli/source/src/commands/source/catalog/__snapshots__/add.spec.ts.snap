// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source:catalog:add when the batch upload fails returns a summary even if an error is thrown 1`] = `
"=== Push Summary

0/10 document successfully sent to the API
2/10 documents failed to be sent
8/10 unprocessed documents

"
`;

exports[`source:catalog:add when the batch upload fails returns an information message on add failure from the API 1`] = `
"Error while trying to add document.
Response body:
{"status":412,"title":"BAD_REQUEST","detail":"this is a bad request and you should feel bad"}"
`;

exports[`source:catalog:add when the batch upload is successful throws when no flags are specified 1`] = `"You must set the \`files\` flag. Use \`source:catalog:add --help\` to get more information."`;

exports[`source:catalog:add when the batch upload is successful when the source does not contain items should not show a error message when using the --skipFullUploadCheck flag 1`] = `
"Processing files... âœ”
"
`;

exports[`source:catalog:add when the batch upload is successful when the source does not contain items should show error message during document update 1`] = `
"No items detected for this source at the moment.
As a best practice, we recommend doing a full catalog upload by appending --fullUpload to your command.
If you are still getting this message despite having already performed a full catalog upload, append --forceUpdate to your command to discard this message."
`;

exports[`source:catalog:add when the batch upload is successful when uploading 1 documents should print the appropriate # of documents 1`] = `
"Success: 1 document accepted by the API from filename0.Status code: 202 ðŸ‘Œ
    
=== Push Summary

1/1 document successfully sent to the API

"
`;

exports[`source:catalog:add when the batch upload is successful when uploading 2 documents should print the appropriate # of documents 1`] = `
"Success: 2 documents accepted by the API from filename0, filename1.Status code: 202 ðŸ‘Œ
    
=== Push Summary

2/2 documents successfully sent to the API

"
`;

exports[`source:catalog:add when the batch upload is successful when uploading 3 documents should print the appropriate # of documents 1`] = `
"Success: 3 documents accepted by the API from filename0, filename1, filename2.Status code: 202 ðŸ‘Œ
    
=== Push Summary

3/3 documents successfully sent to the API

"
`;

exports[`source:catalog:add when the batch upload is successful when uploading 4 documents should print the appropriate # of documents 1`] = `
"Success: 4 documents accepted by the API from filename0, filename1, filename2, filename3.Status code: 202 ðŸ‘Œ
    
=== Push Summary

4/4 documents successfully sent to the API

"
`;

exports[`source:catalog:add when the batch upload is successful when uploading 5 documents should print the appropriate # of documents 1`] = `
"Success: 5 documents accepted by the API from filename0, filename1, filename2, filename3, filename4.Status code: 202 ðŸ‘Œ
    
=== Push Summary

5/5 documents successfully sent to the API

"
`;

exports[`source:catalog:add when the batch upload is successful when uploading 6 documents should print the appropriate # of documents 1`] = `
"Success: 6 documents accepted by the API from filename0, filename1, filename2, filename3, filename4 and 1 more ....Status code: 202 ðŸ‘Œ
    
=== Push Summary

6/6 documents successfully sent to the API

"
`;
